include/net/socket.h

```diff
@@ -97,26 +97,68 @@ struct zsock_addrinfo {
 	char _ai_canonname[DNS_MAX_NAME_SIZE + 1];
 };
 
-int zsock_socket(int family, int type, int proto);
-int zsock_close(int sock);
-int zsock_bind(int sock, const struct sockaddr *addr, socklen_t addrlen);
-int zsock_connect(int sock, const struct sockaddr *addr, socklen_t addrlen);
-int zsock_listen(int sock, int backlog);
-int zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen);
-ssize_t zsock_send(int sock, const void *buf, size_t len, int flags);
-ssize_t zsock_recv(int sock, void *buf, size_t max_len, int flags);
-ssize_t zsock_sendto(int sock, const void *buf, size_t len, int flags,
-		     const struct sockaddr *dest_addr, socklen_t addrlen);
-ssize_t zsock_recvfrom(int sock, void *buf, size_t max_len, int flags,
-		       struct sockaddr *src_addr, socklen_t *addrlen);
-int zsock_fcntl(int sock, int cmd, int flags);
-int zsock_poll(struct zsock_pollfd *fds, int nfds, int timeout);
+/*
+ * NOTE NOTE NOTE NOTE
+ *
+ * System calls have been added at the socket API level so that we can
+ * provide user mode access to the network stack fairly quickly.
+ *
+ * In the fullness of time, the system call layer should be moved much
```

Comment:
I wouldn't leave this note in the version to merge. Nor I'm sure the statement ("In the fullness of time, the system call layer should be moved much further down the stack") is universally right. How down is "much further down"?, etc. Better not leave such prophecies in the source code. Could (should?) go into the commit message though.

Reply:
(Of course, if you mean "syscalls" like getaddrinfo and inet_pton - yeah, those could live in userspace completely. The core stuff like socket/send/recv are pretty much kernel stuff IMHO. Btw, chosen TLS support implementation complicates that too.)